store.cpp

```diff
@@ -6,10 +6,38 @@
 
 #include <glog/logging.h>
 
+Store::Store(std::optional<std::string>& filename) {
+  if (filename != std::nullopt) {
+    std::ifstream fin;
+    std::string line;
+    std::string key;
+    std::string val;
+    fin.open(filename.value());
+    while (fin) {
+      getline(fin, line);
+      int pos = line.find(':');
```

Comment:
I believe you're trying to use the colon as a delimiter? What if the key or the value has a colon in it? Alternatively, you could consider using C++ serialization/deserialization and it might also simplify your parsing code. JSON could also be an option.

Reply:
Yes, I am using colon as a delimiter. It does not matter if the value has a colon because I only split at the first colon found, which means the colon in the value will be parsed into value. As for the key, you are right. I do this simply because I have full control of all keys put into the kvstore and they do not have colons. By using C++ serialization/deserialization, you mean "<<" and ">>" or something else? I agree that using "fin >> line" will make the code a little bit simpler. By using JSON, a good point, I will do some googling. Thanks.

Reply:
(1) It seems using ">>" with a file stream get word by word rather than line by line from the stream
(2) using JSON means other third-party libraries

Reply:
By serialization, I meant more like Boost's https://www.boost.org/doc/libs/1_72_0/libs/serialization/doc/index.html. Native serialization is not available in C++ due to its lack of reflection.

But I agree that you will need a third party library either way and the way you do it is fine. 

As for delimiting on the colon, as long as you ensure your key does not have a colon, it makes sense to me. Thanks!